### DISCUSSION TITLE
Update from v.0.11.0 to v.1.0.0 is not working

### DISCUSSION THREAD

**Discussion by SirAdam2k** [Luligu/matterbridge-shelly#57] [Q&A]
Hello there,

I installed matterbridge via docker compose and matterbridge is running. I added the matterbridge-shelly Plugin and saw that it needs to be Updates to Version v.1.0.0 but after pressing the button for the update and restarting the matterbridge, the update is still showing. Anything I could do wrong?

Thanks and best regards
SirAdam

**Comment by Luligu** [comment]
Hi,
I need a full log to answer, but in general with docker using the official image there is no problem updating.
Can you provide a full log please?
https://github.com/Luligu/matterbridge?tab=readme-ov-file#how-to-send-the-debug-log-files

**Reply by SirAdam2k** [reply]
Hi,

please find the Logs attached.

[matterbridge.log](https://github.com/user-attachments/files/17336956/matterbridge.log)
[matter.log](https://github.com/user-attachments/files/17336960/matter.log)

And this is my docker compose file:

`version: "3.4"

services:
  matterbridge:
    container_name: matterbridge
    image: luligu/matterbridge:latest # Matterbridge image with the latest tag
    network_mode: host # Ensures the Matter mdns works
    restart: always # Ensures the container always restarts automatically
    volumes:
      - /share/swarm/appdata/Matterbridge:/root/Matterbridge # Mounts the Matterbridge plugin directory
      - /share/swarm/appdata/.matterbridge:/root/.matterbridge # Mounts the Matterbridge storage directory
    networks:
      - host
    deploy:
      labels:
        - "traefik.enable=true"
        - "traefik.http.routers.matterbridge.entrypoints=https"
        - "traefik.http.routers.matterbridge.rule=Host(`matterbridge.???.xyz`)"
        - "traefik.http.routers.matterbridge.service=matterbridge"
        - "traefik.http.routers.matterbridge.tls.certresolver=cloudflare"
        - "traefik.http.routers.matterbridge.middlewares=forward-auth@file"
        - "traefik.http.services.matterbridge.loadbalancer.server.scheme=http"       
        - "traefik.http.services.matterbridge.loadbalancer.server.port=8283"

networks:
  host:
    external: true `

Best regards.

**Comment by Luligu** [comment]
The update went perfect: 
[02:40:02.204] [Matterbridge] [debug] Child process "npm install -g matterbridge-shelly --omit=dev --verbose" exited with code 0 and signal null 
[02:40:02.215] [Matterbridge] [notice] Package matterbridge-shelly   installed correctly 
[02:40:02.244] [Matterbridge] [debug] Child process "npm install -g matterbridge-shelly --omit=dev --verbose" closed with code 0 and signal null 
[02:40:02.252] [Matterbridge] [info] Plugin matterbridge-shelly installed. Full restart required. 

You just need to restart.

For the docker compose file I cannot tell cause it depends on your system that seems complex.

**Reply by SirAdam2k** [reply]
Yes, that were my thoughts as well. I already restarted matterbridge multiple times as well as the docker container but the update button for the plug-in is still showing up.

**Comment by Luligu** [comment]
After the restart, try to refresh the page in the browser cause it may take a while to update the version.

Also keep in mind that if your system recreates the container the plugins will get back to the version in the image that, in this moment, is not up to date cause I need to finalize matterbridge 1.6.0.

**Reply by SirAdam2k** [reply]
Okay, I see. Already used a different device to check if it is only cache that is showing wrong information. Any other way to check which version is running?

**Reply by Luligu** [reply]
When matterbridge restarts the info on the actual versions are up to date.
If the frontend is opened it can take time to update the screen (it should not I know).
Just refresh the page on the browser.

**Reply by SirAdam2k** [reply]
Even hours after the update, it stills shows the old version. Same for the config page. Please see attached photo. ![image](https://github.com/user-attachments/assets/686e8a42-7b74-4744-8698-41b2c1614293)

**Comment by Luligu** [comment]
I imagine that there is some issue in the way your setup manage docker containers.
It seems like is recreated each time.
Try this to better understand:
- update the plugin and wait to see the message that the plugin is updated;
- open a shell into the container: docker exec -it matterbridge /bin/bash
- check the version inside: npm list -g
- restart matterbridge from the frontend;
- reopen the shell and check again...

**Reply by SirAdam2k** [reply]
I get the following error message:

/usr/local/bin/npm: npm: line 2: syntax error near unexpected token `'../lib/cli.js''

/usr/local/bin/npm: npm: line 2: `require('../lib/cli.js')(process)'

**Comment by Luligu** [comment]
Doing npm list -g?
Do you mind joining discord group that is easy to talk in real time?

https://discord.gg/raWywpD4

**Reply by SirAdam2k** [reply]
The link doesn‘t seem to work. Discord tells mit it is not valid.

**Comment by Luligu** [comment]
![image](https://github.com/user-attachments/assets/64c00abf-028f-4072-9898-d604bbd31c02)

This is what you should see...

![image](https://github.com/user-attachments/assets/c6fead91-8a14-437a-adea-63701da88584)

**Comment by Luligu** [comment]
https://discord.gg/mmwPdQGJ
